bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/WemoHandlerFactory.java

```diff
@@ -93,6 +99,11 @@ public void unsetWemoHttpCallFactory(WemoHttpCallFactory wemoHttpCallFactory) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
         logger.debug("Trying to create a handler for ThingType '{}", thingTypeUID);
 
+        UDN remoteUDN = new UDN((thing.getConfiguration().get(UDN).toString()));
+        RemoteDevice device = upnpService.getRegistry().getRemoteDevice(remoteUDN, true);
+        URL descriptorURL = device.getIdentity().getDescriptorURL();
```

Comment:
`getRemoteDevice` returns `null` if the device is not found. This can't be detected by the null-checker because `UpnpService` has no null annotations. This results in an NPE in l. 104.

Reply:
Thanks @J-N-K That's what I guessed as well and tried to surrond it with null checks, which was not working. Will try to solve this later today or tomorrow....